src/test/java/plugins/DashboardViewPluginTest.java

```diff
@@ -34,6 +36,58 @@ public void configure_dashboard() {
     }
 
     @Test
+    public void unstableJobsPortlet_failedJob() {
+        DashboardView v = createDashboardView();
+        UnstableJobsPortlet unstableJobsPortlet = v.addBottomPortlet(UnstableJobsPortlet.class);
+        v.save();
+
+        FreeStyleJob j = createFailingFreeStyleJob();
+        buildFailingJob(j);
+
+        v.open();
+
+        assertThat(unstableJobsPortlet.hasJob(j.name), is(true));
+
+        unstableJobsPortlet.openJob(j.name);
+
+        assertThat(driver, hasContent("Project " + j.name));
+        assertThat(getCurrentUrl().contains(j.name), is(true));
+    }
+
+    @Test
+    public void unstableJobsPortlet_successfulJob() {
+        DashboardView v = createDashboardView();
+        UnstableJobsPortlet unstableJobsPortlet = v.addBottomPortlet(UnstableJobsPortlet.class);
+        v.save();
+
+        FreeStyleJob j = createFreeStyleJob();
+        buildSuccessfulJob(j);
+
+        v.open();
+
+        assertThat(unstableJobsPortlet.hasJob(j.name), is(false));
+    }
+
+    @Test
+    public void unstableJobsPortlet_unstableJob() {
+        DashboardView v = createDashboardView();
+        UnstableJobsPortlet unstableJobsPortlet = v.addBottomPortlet(UnstableJobsPortlet.class);
+        v.save();
+
+        FreeStyleJob j = createUnstableFreeStyleJob();
+        buildUnstableJob(j);
+
+        v.open();
+
+        assertThat(unstableJobsPortlet.hasJob(j.name), is(true));
+
+        unstableJobsPortlet.openJob(j.name);
+
+        assertThat(driver, hasContent("Project " + j.name));
+        assertThat(getCurrentUrl().contains(j.name), is(true));
+    }
+
```

Comment:
These three tests look almost the same. I wonder if it would make sense to extract the boilerplate code into a method that accepts the job creation as a lambda? The assertion could be given as boolean flag.